File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let map={}
        for(const num of nums){
            if(map[num]){
                map[num]+=1
            }
            else{
                map[num]=1
            }
        }
        console.log(map)
        // sort values descending
        // slice k values and return keys
        let sortedObject=Object.entries(map).sort(([,a],[,b])=>b-a)
        let finalAnswer=sortedObject.slice(0,k)
        return finalAnswer.map((entry)=>entry[0])  
    

    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        summ = sum(nums)
        if summ%2 == 1:
            return False
        dp = set()
        dp.add(0)
        i = len(nums)-1
        while i>=0:
            newPossible = set(dp)
            for num in dp:
                newPossible.add(num+nums[i])
            dp = newPossible
            i-=1
        return summ//2 in dp